okr_app/tests.py

```diff
@@ -1,4 +1,134 @@
+        """
+        self.client.login(username='guest', password='12345')
+        self.url = '/api/okrfiles/?okr={okr}'.format(okr=self.okr.id) 
+        response = self.client.get(self.url, {}, format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        """
+        Equal to 0 because user is not OKR issuer
+        """
+        self.assertEqual(len(response.data), 0)
+
+    def test_get_okr_file_list_without_logged_in_user(self):
+        """
+        Non authenticated user can't get list of files of a single OKR
+        """
+        self.url = '/api/okrfiles/?okr={okr}'.format(okr=self.okr.id) 
+        response = self.client.get(self.url, {}, format='json')
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_get_okr_file_detail_with_okr_owner(self):
+        """
+        OKR issuer can get OKR's uploaded file
+        """
+        self.client.login(username='okrowner', password='12345')
+        self.url = '/api/okrfiles/{id}/'.format(id=self.okr_file.id) 
+        response = self.client.get(self.url, {}, format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["id"], self.okr_file.id)
+
+    def test_get_okr_file_detail_with_non_okr_owner(self):
+        """
+        Other regular user can't get each other OKR's file
+        """
+        self.client.login(username='guest', password='12345')
+        self.url = '/api/okrfiles/{id}/'.format(id=self.okr_file.id) 
+        response = self.client.get(self.url, {}, format='json')
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_get_okr_file_detail_without_logged_in_user(self):
+        """
+        Non authenticated user can't get each other OKR's file
+        """
+        self.url = '/api/okrfiles/{id}/'.format(id=self.okr_file.id) 
+        response = self.client.get(self.url, {}, format='json')
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_delete_with_non_okr_owner(self):
+        """
+        Other regular user can't delete each other OKR's file
+        """
+        self.client.login(username='guest', password='12345')
+        self.url = '/api/okrfiles/{id}/'.format(id=self.okr_file.id) 
+        response = self.client.delete(self.url, {}, format='json')
+        """
+        Not found because queryset is filtered by issuer
+        """
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    #Can ?
+    def test_delete_with_okr_owner(self):
+        """
+        OKR issuer can delete OKR's uploaded file
+        """
+        self.client.login(username='okrowner', password='12345')
+        self.url = '/api/okrfiles/{id}/'.format(id=self.okr_file.id) 
+        response = self.client.delete(self.url, {}, format='json')
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
```

Comment:
use coverage to make sure that your tests cover all the code
https://coverage.readthedocs.io/en/6.2/

Reply:
![coverage](https://user-images.githubusercontent.com/62880003/146535197-1e7084e6-c654-47a0-bc46-db1015777451.png)